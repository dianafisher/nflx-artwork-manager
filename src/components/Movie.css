.small-thumbnails {
  display: flex;
  flex-direction: row;
  flex-wrap: wrap;
}

.movie {
  margin-bottom: 20px;
}

.movie-title {
  margin-left: 10px;
  font-size: 30px;
  text-align: left;
}
