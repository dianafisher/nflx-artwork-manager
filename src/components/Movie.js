import React from 'react';
import PropTypes from 'prop-types';
import SmallThumbnail from './SmallThumbnail';
import './Movie.css';

/*
  A Movie component displays a SmallThumbnail component for each entry in
  it's group.

  This component passes the onThumbnailClicked prop down to the SmallThumbnail
*/
function Movie (props) {
  // If we do not have data for some reason, do not render anything.
  if (props.entries.length === 0) {
    return null;
  }

  const languageCode = props.entries[0].languageCode;
  const movieName = props.entries[0].movieName;

  const langTitle = `${props.entries[0].language} (${languageCode})` ;

  const title = props.groupedBy === 'movieId' ? movieName : langTitle;

  return (
    <div className='movie'>
      <div className='movie-title'>{title}</div>
      <div className='small-thumbnails'>
        {props.entries.map((entry) =>
          <SmallThumbnail
            key={entry.thumbnailUrl}
            onClick={props.onThumbnailClicked}
            data={entry}
          ></SmallThumbnail>
        )}
      </div>
    </div>
  )
}

Movie.propTypes = {
  onThumbnailClicked: PropTypes.func.isRequired,
  groupedBy: PropTypes.string.isRequired,
  entries: PropTypes.array.isRequired,
};

export default Movie;
