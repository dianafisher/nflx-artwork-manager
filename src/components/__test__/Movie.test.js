import React from 'react';
import { shallow } from 'enzyme';
import Movie from '../Movie';

const groupedBy = 'movieId'
const entries = [{"movieId":70242311,"movieName":"Orange Is the New Black","imageType":"sdp","thumbnailUrl":"http://art.nflximg.net/673e9/b39fcc29b2ac668ee01343de9f21f611c8f673e9.jpg","fullSizeImageUrl":"http://art.nflximg.net/78bc7/198343ed941f178d54878aa366a122e4e2e78bc7.jpg","languageCode":"it"}];

const invalidEntry = {
  'movieId': 70242311,
  "movieName": "Orange Is the New Black",
  "langgeCode": "it",
  "thumbnailUrl":"http://art.nflximg.net/673e9/b39fcc29b2ac668ee01343de9f21f611c8f673e9.jpg",
  "fullSizeImageUrl":"http://art.nflximg.net/78bc7/198343ed941f178d54878aa366a122e4e2e78bc7.jpg"
};

describe('Movie', () => {
  it('renders with an entry', () => {
    expect(shallow(
      <Movie
        onThumbnailClicked={jest.fn()}
        groupedBy={groupedBy}
        entries={entries}
      ></Movie>
    )).toMatchSnapshot();
  });

  it('does not explode with no data', () => {
    expect(shallow(
      <Movie
        onThumbnailClicked={jest.fn()}
        groupedBy={groupedBy}
        entries={[]}
      ></Movie>
    ))
  });

  it('does not explode with invalid data', () => {
    expect(shallow(
      <Movie
        onThumbnailClicked={jest.fn()}
        groupedBy='uknown'
        entries={[invalidEntry]}
      ></Movie>
    ))
  });

});
