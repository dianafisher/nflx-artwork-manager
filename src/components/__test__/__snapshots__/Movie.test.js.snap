// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Movie renders with an entry 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <div
    className="movie"
>
    <div
        className="movie-title"
    >
        Orange Is the New Black
    </div>
    <div
        className="small-thumbnails"
    >
        <SmallThumbnail
            data={
                Object {
                    "fullSizeImageUrl": "http://art.nflximg.net/78bc7/198343ed941f178d54878aa366a122e4e2e78bc7.jpg",
                    "imageType": "sdp",
                    "languageCode": "it",
                    "movieId": 70242311,
                    "movieName": "Orange Is the New Black",
                    "thumbnailUrl": "http://art.nflximg.net/673e9/b39fcc29b2ac668ee01343de9f21f611c8f673e9.jpg",
                  }
            }
            onClick={[Function]}
        />
    </div>
</div>,
  "nodes": Array [
    <div
      className="movie"
>
      <div
            className="movie-title"
      >
            Orange Is the New Black
      </div>
      <div
            className="small-thumbnails"
      >
            <SmallThumbnail
                  data={
                        Object {
                              "fullSizeImageUrl": "http://art.nflximg.net/78bc7/198343ed941f178d54878aa366a122e4e2e78bc7.jpg",
                              "imageType": "sdp",
                              "languageCode": "it",
                              "movieId": 70242311,
                              "movieName": "Orange Is the New Black",
                              "thumbnailUrl": "http://art.nflximg.net/673e9/b39fcc29b2ac668ee01343de9f21f611c8f673e9.jpg",
                            }
                  }
                  onClick={[Function]}
            />
      </div>
</div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 2,
      "_context": Object {},
      "_currentElement": <Movie
        entries={
                Array [
                        Object {
                          "fullSizeImageUrl": "http://art.nflximg.net/78bc7/198343ed941f178d54878aa366a122e4e2e78bc7.jpg",
                          "imageType": "sdp",
                          "languageCode": "it",
                          "movieId": 70242311,
                          "movieName": "Orange Is the New Black",
                          "thumbnailUrl": "http://art.nflximg.net/673e9/b39fcc29b2ac668ee01343de9f21f611c8f673e9.jpg",
                        },
                      ]
        }
        groupedBy="movieId"
        onThumbnailClicked={[Function]}
/>,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": StatelessComponent {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "entries": Array [
            Object {
              "fullSizeImageUrl": "http://art.nflximg.net/78bc7/198343ed941f178d54878aa366a122e4e2e78bc7.jpg",
              "imageType": "sdp",
              "languageCode": "it",
              "movieId": 70242311,
              "movieName": "Orange Is the New Black",
              "thumbnailUrl": "http://art.nflximg.net/673e9/b39fcc29b2ac668ee01343de9f21f611c8f673e9.jpg",
            },
          ],
          "groupedBy": "movieId",
          "onThumbnailClicked": [Function],
        },
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <div
          className="movie"
>
          <div
                    className="movie-title"
          >
                    Orange Is the New Black
          </div>
          <div
                    className="small-thumbnails"
          >
                    <SmallThumbnail
                              data={
                                        Object {
                                                  "fullSizeImageUrl": "http://art.nflximg.net/78bc7/198343ed941f178d54878aa366a122e4e2e78bc7.jpg",
                                                  "imageType": "sdp",
                                                  "languageCode": "it",
                                                  "movieId": 70242311,
                                                  "movieName": "Orange Is the New Black",
                                                  "thumbnailUrl": "http://art.nflximg.net/673e9/b39fcc29b2ac668ee01343de9f21f611c8f673e9.jpg",
                                                }
                              }
                              onClick={[Function]}
                    />
          </div>
</div>,
        "_debugID": 2,
        "_renderedOutput": <div
          className="movie"
>
          <div
                    className="movie-title"
          >
                    Orange Is the New Black
          </div>
          <div
                    className="small-thumbnails"
          >
                    <SmallThumbnail
                              data={
                                        Object {
                                                  "fullSizeImageUrl": "http://art.nflximg.net/78bc7/198343ed941f178d54878aa366a122e4e2e78bc7.jpg",
                                                  "imageType": "sdp",
                                                  "languageCode": "it",
                                                  "movieId": 70242311,
                                                  "movieName": "Orange Is the New Black",
                                                  "thumbnailUrl": "http://art.nflximg.net/673e9/b39fcc29b2ac668ee01343de9f21f611c8f673e9.jpg",
                                                }
                              }
                              onClick={[Function]}
                    />
          </div>
</div>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <Movie
    entries={
        Array [
            Object {
              "fullSizeImageUrl": "http://art.nflximg.net/78bc7/198343ed941f178d54878aa366a122e4e2e78bc7.jpg",
              "imageType": "sdp",
              "languageCode": "it",
              "movieId": 70242311,
              "movieName": "Orange Is the New Black",
              "thumbnailUrl": "http://art.nflximg.net/673e9/b39fcc29b2ac668ee01343de9f21f611c8f673e9.jpg",
            },
          ]
    }
    groupedBy="movieId"
    onThumbnailClicked={[Function]}
/>,
}
`;
